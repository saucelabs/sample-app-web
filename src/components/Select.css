.select_container {
  box-sizing: border-box;
  background-color: #ffffff;
  overflow: hidden;
  padding: 0 0 0 40px;
  cursor: pointer;
  height: 40px;
  width: 40px;
  display: block;
  position: relative;
}

.active_option {
  display: none;
}

.select_container:before {
  content: "";
  background: url("../assets/img/filter.png") no-repeat center center;
  background: url("../assets/svg/filter3x.svg") no-repeat center center,
    linear-gradient(transparent, transparent);
  height: 40px;
  width: 40px;
  position: absolute;
  left: 0;
  top: 0;
}

.product_sort_container {
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  width: 100%;
  /* This needs to be like this because otherwise Safari can't click on an element that is not visible */
  opacity: 0.001;
}

@media only screen and (min-width: 900px) {
  .select_container {
    padding: 0 40px;
    width: 250px;
    line-height: 40px;
  }

  .select_container:after {
    content: "";
    background: url("../assets/img/arrow.png") no-repeat center center;
    background: url("../assets/svg/arrow3x.svg") no-repeat center center,
      linear-gradient(transparent, transparent);
    height: 40px;
    width: 40px;
    position: absolute;
    right: 0;
    top: 0;
    transform: rotate(270deg);
  }

  .active_option {
    display: block;
    font-size: 16px;
    text-align: center;
    text-transform: uppercase;
  }
}
