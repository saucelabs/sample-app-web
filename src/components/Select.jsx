import React from "react";
import PropTypes from "prop-types";
import "./Select.css";

const Select = ({ activeOption, onChange, options, testId }) => {
  return (
    <span className="select_container">
      <span className="active_option">
        {
          options[options.findIndex((option) => option.key === activeOption)]
            .value
        }
      </span>
      <select
        onChange={onChange}
        className="product_sort_container"
        value={activeOption}
        {...(testId
          ? {
              "data-test": testId,
            }
          : {})}
      >
        {options.map(({ key, value }) => (
          <option value={key} key={key}>
            {value}
          </option>
        ))}
      </select>
    </span>
  );
};
Select.propTypes = {
  /**
   * The active option key
   */
  activeOption: PropTypes.string.isRequired,
  /**
   * The on change handler
   */
  onChange: PropTypes.func.isRequired,
  /**
   * The options
   */
  options: PropTypes.arrayOf(
    PropTypes.shape({
      key: PropTypes.string.isRequired,
      value: PropTypes.string.isRequired,
    })
  ).isRequired,
  /**
   * The test id
   */
  testId: PropTypes.string,
};
Select.defaultProps = {
  testId: undefined,
};

export default Select;
