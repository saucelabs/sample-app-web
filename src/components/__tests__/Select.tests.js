import React from "react";
import { shallow } from "enzyme";
import Select from "../Select";

let props;

describe("Select", () => {
  beforeEach(() => {
    props = {
      activeOption: "az",
      options: [
        { key: "az", value: "Name (A to Z)" },
        { key: "za", value: "Name (Z to A)" },
        { key: "lohi", value: "Price (low to high)" },
        { key: "hilo", value: "Price (high to low)" },
      ],
      onChange: jest.fn(),
    };
  });

  it("should render with default props", () => {
    const component = shallow(<Select {...props} />);

    expect(component).toMatchSnapshot();
  });

  it("should render with a testID", () => {
    const component = shallow(<Select testId="foo" {...props} />);

    expect(component).toMatchSnapshot();
  });

  it("should be able to trigger the onChange", () => {
    const component = shallow(<Select {...props} />);
    component
      .find("select")
      .at(0)
      .simulate("change", {
        target: { value: "za", name: "Name (Z to A)" },
      });

    expect(props.onChange).toHaveBeenCalledTimes(1);
  });
});
