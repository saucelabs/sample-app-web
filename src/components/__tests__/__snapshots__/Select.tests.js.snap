// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select should render with a testID 1`] = `
<span
  className="select_container"
>
  <span
    className="active_option"
  >
    Name (A to Z)
  </span>
  <select
    className="product_sort_container"
    data-test="foo"
    onChange={[MockFunction]}
    value="az"
  >
    <option
      key="az"
      value="az"
    >
      Name (A to Z)
    </option>
    <option
      key="za"
      value="za"
    >
      Name (Z to A)
    </option>
    <option
      key="lohi"
      value="lohi"
    >
      Price (low to high)
    </option>
    <option
      key="hilo"
      value="hilo"
    >
      Price (high to low)
    </option>
  </select>
</span>
`;

exports[`Select should render with default props 1`] = `
<span
  className="select_container"
>
  <span
    className="active_option"
  >
    Name (A to Z)
  </span>
  <select
    className="product_sort_container"
    onChange={[MockFunction]}
    value="az"
  >
    <option
      key="az"
      value="az"
    >
      Name (A to Z)
    </option>
    <option
      key="za"
      value="za"
    >
      Name (Z to A)
    </option>
    <option
      key="lohi"
      value="lohi"
    >
      Price (low to high)
    </option>
    <option
      key="hilo"
      value="hilo"
    >
      Price (high to low)
    </option>
  </select>
</span>
`;
